src/main/cpp/src/decimal_utils.cu

```diff
@@ -675,21 +677,24 @@ struct dec128_multiplier {
 
     chunked256 product = multiply(a, b);
 
+    int dec_precision = precision10(product);
+
+    int mult_scale = a_scale + b_scale;
+
     // Spark does some really odd things that I personally think are a bug
```

Comment:
nit: Could we update this comment to explain what some versions of Spark do this and it has been fixed.